javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,39 @@
+var restart = document.querySelector('button');
+var typer = document.getElementById('phrase');
+var arr = typer.textContent.split('');
+var spanArr = [];
+var position =  0;
+
+for(var i = 0; i < arr.length; i++){
+  var spanCharacter = document.createElement('span');
+  spanCharacter.textContent = arr[i];
+  spanArr.push(spanCharacter);
+}
+
+window.addEventListener('keyup', function(e) {
+  if (arr[position].toLowerCase() === e.key) {
+
+    spanArr[position].classList.remove('incorrect');
+    spanArr[position].classList.add('correct');
+
+    spanArr[position].classList.remove('underline');
+    spanArr[position+1].classList.add('underline');
+
+    typer.innerHTML = '';
+
+    for(var i = 0; i < arr.length; i++){
+      typer.appendChild(spanArr[i]);
+    }
+
+    position++
+
+  }else{
+    spanArr[position].classList.add('incorrect');
+
+  }
+
+});
+
+restart.addEventListener('click', function(e) {
+  window.location.reload();
+})
```

Comment:
You should **_never_** use `location.reload`. It entirely refreshes the page, so the whole page needs to get built again from scratch, which is highly inefficient. Rather, if somebody wants to play again, you just remove all the classes from the letters that are already applied, and reset the value of your variable for what `position` the user is at, so they can just play again without needing to refresh.